lib/Widgets/Placeholder.vala

```diff
@@ -0,0 +1,167 @@
+        set_layout_manager_type (typeof (Gtk.BinLayout));
+    }
+
+    construct {
+        var image = new Gtk.Image () {
+            icon_size = Gtk.IconSize.LARGE,
+            valign = Gtk.Align.START
+        };
+
+        var title_label = new Gtk.Label (title) {
+            max_width_chars = 30,
+            wrap = true,
+            xalign = 0
+        };
+        title_label.add_css_class (Granite.STYLE_CLASS_H1_LABEL);
+
+        var description_label = new Gtk.Label ("") {
+            max_width_chars = 45,
+            wrap = true,
+            use_markup = true,
+            xalign = 0
+        };
+        description_label.add_css_class (Granite.STYLE_CLASS_H2_LABEL);
+
+        buttonbox = new Gtk.Box (Gtk.Orientation.VERTICAL, 0) {
+            visible = false
+        };
+
+        var grid = new Gtk.Grid ();
+        grid.attach (image, 0, 0, 1, 2);
+        grid.attach (title_label, 1, 0);
+        grid.attach (description_label, 1, 1);
+        grid.attach (buttonbox, 1, 2);
+        grid.set_parent (this);
+
+        halign = Gtk.Align.CENTER;
+        valign = Gtk.Align.CENTER;
+        hexpand = true;
+
+        bind_property ("title", title_label, "label");
+        bind_property ("description", description_label, "label");
+        bind_property (
+            "description", description_label, "visible", BindingFlags.SYNC_CREATE | BindingFlags.DEFAULT,
+            (binding, srcval, ref targetval) => {
+                var label = (string) srcval;
+                if (label != null) {
+                    targetval.set_boolean (true);
+                } else {
+                    targetval.set_boolean (false);
+                }
+
+                return true;
+            },
+            null
+        );
+
+        bind_property ("icon", image, "gicon");
+        bind_property (
+            "icon", image, "visible", BindingFlags.SYNC_CREATE | BindingFlags.DEFAULT,
+            (binding, srcval, ref targetval) => {
+                var gicon = (Icon) srcval;
+                if (gicon != null) {
+                    targetval.set_boolean (true);
+                } else {
+                    targetval.set_boolean (false);
+                }
+
+                return true;
+            },
+            null
+        );
+    }
+
+    ~Placeholder () {
+        get_first_child ().unparent ();
+    }
+
+    /**
+     * Appends new {@link Gtk.Button} to the placeholder's action area
+     *
+     * @param icon the {@link GLib.Icon} that describes this action
+     * @param label a {@link string} to use as the title for this action. It should use Title Case.
+     * @param description a {@link string} to use as a description for this action. It should use sentence case.
+     * @return a {@link Gtk.Button} representing this action
+     */
+    public Gtk.Button append_button (Icon icon, string label, string description) {
+        var image = new Gtk.Image.from_gicon (icon) {
+            icon_size = Gtk.IconSize.LARGE
```

Comment:
Any reason to set large on every image? The only thing it does is adding `.large` style class, as sizes are now defined in css.

Reply:
Yeah I could just not set it here and rely totally on setting it in our stylesheet, but this gets me 32px in `default` as well so it's slightly less broken even without actual support in Adwaita

Reply:
Ok.